### PR TITLE
Fix problem with passing secrets for Smoke tets job

### PR DESCRIPTION
**Pull request by musienko-maxim**
This PR fixes the problem of passing secrets between jobs. More details were described in the issue: https://github.com/eclipse/che/issues/22250
The main idea build and prepare images for verification in one workflow; It was checked by running PR check directly from this branch in the  **developer-image** project and from my fork examples:
* from the  personal fork https://github.com/devfile/developer-images/actions/runs/5455491650
* from this branch we can see the results below

### CODE REVIEW

File: .github/workflows/empty-worksapce-smoke-test-on-minikube-ubi8.yaml [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+      run: |
+        docker push quay.io/devfile/universal-developer-image:${{ env.PR_NUMBER }}
+
+    - name: Start minikube cluster
+      id: run-minikube
+      uses: che-incubator/setup-minikube-action@next
+      with:
+        minikube-version: v1.23.2
+
+    - name: Checkout WTO
```
Comment (l0rd) [inline .github/workflows/empty-worksapce-smoke-test-on-minikube-ubi8.yaml:66]:
```suggestion
    - name: Checkout DWO
```

File: .github/workflows/empty-worksapce-smoke-test-on-minikube-ubi8.yaml [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+
+    - name: Push universal base image
+      run: |
+        docker push quay.io/devfile/universal-developer-image:${{ env.PR_NUMBER }}
+
+    - name: Start minikube cluster
+      id: run-minikube
+      uses: che-incubator/setup-minikube-action@next
+      with:
+        minikube-version: v1.23.2
```
Comment (l0rd) [inline .github/workflows/empty-worksapce-smoke-test-on-minikube-ubi8.yaml:64]: v1.23.2 is almost 2 years old. Is there any concern using the latest ([v1.30.1](https://github.com/kubernetes/minikube/releases/tag/v1.30.1))?

```suggestion
        minikube-version: v1.30.1
```

**Review by l0rd — COMMENTED** [review]
I provided a couple of suggestions. Apart from that this is good @musienko-maxim.

**Review by l0rd — APPROVED** [review]
LGTM. Thanks @musienko-maxim for addressing that. I can merge this PR as soon as you are ready.

**Comment by musienko-maxim** [comment]
I think we can merge it

**Pull request merged**